### PR TITLE
 Explaining autocast in release post

### PR DESCRIPTION
**Pull request by beta-ziliani**
Completes a section of #219

### CODE REVIEW

File: _releases/2022-01-06-1.3.0-released.md [hunk 1]
```diff
@@ -24,6 +24,40 @@ In this release we included [XXX PRs since the 1.2.2 release](https://github.com
 
 ## Number Autocast
 
+Numeric elements are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32`:
```
Comment (straight-shoota) [inline _releases/2022-01-06-1.3.0-released.md:27]:
```suggestion
Primitive numeric values are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32`:
```

File: _releases/2022-01-06-1.3.0-released.md [hunk 2]
```diff
@@ -24,6 +24,40 @@ In this release we included [XXX PRs since the 1.2.2 release](https://github.com
...
+<div class="code_section">{% highlight crystal %}
+def foo(x : Int64)
+  x
+end
+
+def bar
+  1_i32
+end
+
+foo(bar)
```
Comment (straight-shoota) [inline _releases/2022-01-06-1.3.0-released.md:38]: I don't think there's a need to have a method here?

```suggestion
bar = 1_i32
foo(bar)
```

File: _releases/2022-01-06-1.3.0-released.md [hunk 3]
```diff
@@ -24,6 +24,40 @@ In this release we included [XXX PRs since the 1.2.2 release](https://github.com
...
+{% endhighlight %}</div>
+
+Similarly, unsigned integer types can be casted into larger signed ones, and it also works for `Float32` (to be casted as `Float64`).
```
Comment (straight-shoota) [inline _releases/2022-01-06-1.3.0-released.md:41]:
```suggestion
Unsigned integer types can be autocasted into larger signed ones. And autocasting also works for floating point types (`Float32` to `Float64`).
```

File: _releases/2022-01-06-1.3.0-released.md [hunk 4]
```diff
@@ -24,6 +24,40 @@ In this release we included [XXX PRs since the 1.2.2 release](https://github.com
...
+  x
+end
+
+def foo(x : Int128)
+  x
+end
+
+def bar
+  1_i32
+end
```
Comment (straight-shoota) [inline _releases/2022-01-06-1.3.0-released.md:56]:
```suggestion
bar = 1_i32
```

**Review by straight-shoota — COMMENTED** [review]
I'd add "Previously, autocasting only worked for number literals." somwehere. Maybe after the first example?

**Comment by straight-shoota** [comment]
WDYT about this:
> I'd add "Previously, autocasting only worked for number literals." somwehere. Maybe after the first example?

**Comment by beta-ziliani** [comment]
Sorry, I missed that one

**Comment by straight-shoota** [comment]
Typing bird would've picked it up 😆

**Comment by beta-ziliani** [comment]
And that's why AI will never succeed in overthrowing us 😏

File: _releases/2022-01-06-1.3.0-released.md [hunk 5]
```diff
@@ -24,15 +24,17 @@ In this release we included [XXX PRs since the 1.2.2 release](https://github.com
...
 
-Primitive numeric values are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32`:
+Primitive numeric values are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32` (note that before 1.3.0 only number _literals_ were autocasted):
 
 <div class="code_section">{% highlight crystal %}
 def foo(x : Int64)
   x
 end
 
+foo 1_32 # Works in 1.2.2
```
Comment (straight-shoota) [inline _releases/2022-01-06-1.3.0-released.md:34]:
```suggestion
foo 1_32 # Works in 1.2.2 and 1.3.0
```

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**